kong/dynamic_hook/wrap_function_gen.lua

```diff
@@ -0,0 +1,221 @@
+local ngx_get_phase = ngx.get_phase
+
+local TEMPLATE = [[
+  return function(always_enabled_groups, group_name, original_func, handlers)
+    local ngx_get_phase = ngx.get_phase
```

Comment:
duplicated line: `local ngx_get_phase = ngx.get_phase`

Reply:
It is required, this template function can not catch the upvalue.

Reply:
Please a comment to let others know it.

Reply:
done